### ISSUE TITLE
duplicate functions in imf_version.json

### ISSUE THREAD

**Issue by luke-titley** [vfx-rs/cppmm#57]
I'm getting duplicate outputs for functions inside of bind/imf_version.cpp.

To reproduce:
* delete everything from imf_version.cpp apart from isTiled function declaration.
* delete all the bind files except imf_version.cpp and imf_tiledinputpart.cpp
* delete everything in test/openexr/ref/ast
* run astgen to produce imf_version.json and imf_tiledinputpart.json in ref/ast.
* notice that imf_version.json contains two entries for isTiled.

Note that ImfTiledInputPart.h indirectly includes ImfVersion.h (via  ImfTiledInputFile.h -> ImfTiledOutputFile.h -> ImfGenericOutputFile.h)

The log output I get is this:
```
/Volumes/src/cppmm/build /Volumes/src/cppmm
          astgen.cpp: 176 [debug] Found binding file ../test/openexr/bind/imf_version.cpp
          astgen.cpp: 176 [debug] Found binding file ../test/openexr/bind/imf_tiledinputpart.cpp
 process_binding.cpp:1698 [debug]     BIND FUNCTION Imf_2_5::isTiled
 process_binding.cpp:1043 [trace]     -> _Bool
 process_binding.cpp:1049 [trace]         version: int
 process_binding.cpp:1724 [debug] Adding binding function Imf_2_5::isTiled(version: int, ) -> _Bool
 process_binding.cpp:1946 [debug] --- finished matching
 process_binding.cpp:1948 [debug]     Imf_2_5::isTiled
 process_binding.cpp:1954 [debug] Adding matcher for function isTiled
 process_binding.cpp:1043 [trace]     -> _Bool
 process_binding.cpp:1049 [trace]         version: int
 process_binding.cpp:1117 [trace]         matching Imf_2_5::isTiled with Imf_2_5::isTiled
 process_binding.cpp:1819 [debug] MATCHED Imf_2_5::isTiled
 process_binding.cpp:1946 [debug] --- finished matching
 process_binding.cpp:1948 [debug]     Imf_2_5::isTiled
 process_binding.cpp:1954 [debug] Adding matcher for function isTiled
 process_binding.cpp:1043 [trace]     -> _Bool
 process_binding.cpp:1049 [trace]         version: int
 process_binding.cpp:1117 [trace]         matching Imf_2_5::isTiled with Imf_2_5::isTiled
 process_binding.cpp:1819 [debug] MATCHED Imf_2_5::isTiled
7
1
```

**Comment by luke-titley** [comment]
I wonder if it's related to the fact that binding_functions seems to be global state. If HandleTranslationUnit is run once after the other. Does this get cleared out?

**Comment by luke-titley** [comment]
ok so in my isolated test case adding
```
binding_functions.clear()
```
as the first line in ProcessBindingConsumer::HandleTranslationUnit , in process_binding.cpp gets rid of the issue for me.
I have no idea what side effects that might have.

**Comment by luke-titley** [comment]
okay I get it.
enum and vars use a set, functions use a vector.

**Comment by luke-titley** [comment]
This was just the same library functions matching repeatedly the same bind function.
Fixed in  #abd87cd567ef7675590d7ba3c39bd7fb23eee561
